lib/logstash/outputs/scalyr.rb

```diff
@@ -500,6 +500,24 @@ def multi_receive(events)
             result.push(multi_event_request)
           end
 
+        rescue Scalyr::Common::Client::PayloadTooLargeError => e
+          # if the payload is too large, we do not retry. we send to DLQ or drop it.
+          exc_data = {
+              :error_class => e.e_class,
+              :url => e.url.to_s,
+              :message => e.message,
+              :batch_num => batch_num,
+              :total_batches => total_batches,
+              :record_count => multi_event_request[:record_count],
+              :payload_size => multi_event_request[:body].bytesize,
+          }
+          exc_data[:code] = e.code if e.code
+          if defined?(e.body) and e.body
+            exc_data[:body] = Scalyr::Common::Util.truncate(e.body, 512)
+          end
+          exc_data[:payload] = "\tSample payload: #{request[:body][0,1024]}..."
+          log_retry_failure(multi_event_request, exc_data, 0, 0)
```

Comment:
I think it would be better to log this under warning instead of error.

In addition to that, we should probably also allowing passing custom message to be logged to ``log_retry_failure`` so we can use something along the lines of "Not retrying non-retriable request" or similar instead of "Failed retrying after 0 retries).

Reply:
In fact, to not alert and potentially confuse the users, I think we should use warning log level for retrying and unexpected error and only use error log level if we fail to retry after exhausting all the retries. That's the same behavior we use on the agent (retry itself is not fatal so warning is a better fit than error).

Reply:
@Kami this is done.